apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -323,65 +437,57 @@ int init_test_file(void)
 		return ERROR;
 	}
 
-	ret = create_file(TEST_LARGE, TEST_LARGE_BUFSIZE);
+	ret = create_file(TEST_LARGE, TEST_LARGE_BUFSIZE, FILE_TYPE_RANDOM);
 	if (ret != OK && errno != EEXIST) {
 		return ERROR;
 	}
 
-	ret = create_file(TEST_LARGE_BACKUP, TEST_LARGE_BUFSIZE);
+	ret = create_file(TEST_LARGE_BACKUP, TEST_LARGE_BUFSIZE, FILE_TYPE_RANDOM);
 	if (ret != OK && errno != EEXIST) {
 		return ERROR;
 	}
 	printf("create test file success!!\n");
 	return OK;
 }
 
-static int do_test(char *src, char *backup, char *backupdir, int bufsize)
+static int do_test(char *src, char *backup, char *backupdir, int bufsize, file_type_t type)
 {
 	int ret;
+	int index;
+	struct stat st;
 	
-	ret = verify_file(src, bufsize);
-	if (ret != TEST_WRITECOUNT) {
-		ret = recovery_file(src, backup, backupdir, bufsize, ret);
-		if (ret != OK) {
-			printf("recovery_file %s failed\n", src);
-			return ERROR;
-		}
-	} else {
-		printf("verify file %s success!! clean backup files\n", src);
-		ret = unlink(backup);
-		if (ret != OK) {
-			printf("unlink %s failed errno : %d\n", backup, errno);
-			ret = ERROR;
-		}
-
-		ret = rmdir(backupdir);
-		if (ret != OK) {
-			printf("rmdir %s failed errno : %d\n", backupdir, errno);
-			ret = ERROR;
-		}
+	ret = verify_file(src, bufsize, &index, type);
+	if (ret == ERROR) {
+		return ret;
 	}
 
+	ret = recovery_file(src, backup, backupdir, bufsize, index, type);
+	if (ret != OK) {
+		printf("recovery_file failed file : %s\n", src);
+		return ERROR;
+	} 
+
 	printf("do_test create backupdir : %s\n", backupdir);
 	ret = mkdir(backupdir, 0777);
 	if (ret != OK) {
-		printf("Test loop, create backup dir for file %s failed errno : %d\n", src, errno);
+		printf("Test loop, create backup dir for file failed dir : %s errno : %d\n", src, errno);
```

Comment:
I think we need to print name "backup" instead of "src" for dir name.

Reply:
You're right, I will fix it.